lib/setupHooks/removeReferencesToRustToolchain.sh

```diff
@@ -0,0 +1,24 @@
+# Strip any references to the rust toolchain (which can slip in through stdlib panic info) to avoid 
+# including the toolchain in the runtime closure
+
+removeReferencesToRustToolchain() {
+    local installLocation="${1:-${out:?not defined}}"
+    # NOTE(adam.dayan): would it be better to explicitly point to $(rustc --print sysroot)/lib/rustlib/src so we only scrub references to
+    # the src? are there any reasons one might want a reference to the rest of the toolchain?
+    echo "stripping references to Rust toolchain"
+    local rustToolchainLocation rustToolchainStoreHash
+    rustToolchainLocation=$(rustc --print sysroot)
+    echo "Rust toolchain at: $rustToolchainLocation"
+    rustToolchainStoreHash=$(echo "$rustToolchainLocation" | rg "^@storeDir@/([a-z0-9]{32})" -or '$1')
+
+    find "${installLocation}" -type f -exec sed -i -E "s|@storeDir@/${rustToolchainStoreHash}([^/]*)|@storeDir@/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee\1|g" {} +;
```

Comment:
Could probably make this simpler by replacing the whole hash and not worrying about capture groups (the odds of a hash collision across packages is pretty astronomical anyway)

Could also amortize the `sed` startup and compilation time by piping through xargs

```suggestion
    find "${installLocation}" -type f | xargs --no-run-if-empty sed -i'' "s|${rustToolchainStoreHash}|@storeDir@/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee|g"
```